publish-to-cgw-wrapper/publish_to_cgw_wrapper.py

```diff
@@ -48,24 +43,77 @@ def parse_args():
         help="The hostname of the content-gateway to publish the metadata to",
     )
     parser.add_argument(
-        "--data_file",
-        required=True,
-        help="Path to the JSON file containing merged data",
-    )
-    parser.add_argument(
-        "--content_dir",
+        "--data_json",
         required=True,
-        help="Path to the directory containing content to push",
+        help="JSON string containing snapshot merged data",
     )
     parser.add_argument(
-        "--output_file",
-        required=True,
-        help="Path to the file which write the result.json file path",
+        "--dry_run",
+        action="store_true",
+        help="Simulate the script without API calls",
     )
 
     return parser.parse_args()
 
 
+def load_data(data_arg):
+    """Load JSON from string"""
+    try:
+        return json.loads(data_arg)
+    except json.JSONDecodeError:
+        raise ValueError("Invalid 'data_json' must be a valid JSON string")
+
+
+def validate_components(data):
+    """
+    Validates snapshot component data. Skips components without a 'contentGateway'
+    and fails if required fields are missing in either the 'contentGateway'
+    or any listed files. Returns only the valid components.
+    """
+    required_cg_keys = ["productCode", "productName", "productVersionName", "contentDir"]
+    required_file_keys = ["filename"]
+    errors = []
+    valid_components = []
+
+    components = data.get("components")
+    if not components:
+        raise ValueError("Missing or empty 'components' in data.")
+
+    for c_num, component in enumerate(components, start=1):
+        if "contentGateway" not in component:
+            logging.warning(
+                f"Configuration is not defined for publishing... "
+                f"skipping component {c_num}"
+            )
+            continue
+
+        error = False
+        for param in required_cg_keys:
+            if not component["contentGateway"].get(param):
+                errors.append(f"Component {c_num} is missing '{param}'")
+                error = True
+
+        for f_num, file in enumerate(component.get("files"), start=0):
+            for param in required_file_keys:
+                if not file.get(param):
+                    errors.append(
+                        f"Component {c_num}, file {f_num} is missing or has empty '{param}'"
+                    )
+                    error = True
+
+        if not error:
+            valid_components.append(component)
+
+    if errors:
+        raise ValueError("Validation failed with the following errors:\n" + "\n".join(errors))
+
+    logging.info(
+        f"Validation summary: {len(components)} total components {len(valid_components)}"
+        f"valid components, {len(components) - len(valid_components)} skipped components"
+    )
```

Comment:
Your fstrings are getting concatenated. So output looks like this: `INFO - Validation summary: 1 total components,  1valid components, 0 skipped components`
```suggestion
    logging.info(
        f"Validation summary: {len(components)} total components, "
        f"{len(valid_components)} valid components, "
        f"{len(components) - len(valid_components)} skipped components"
    )
```

Reply:
fixed ty